### ISSUE TITLE
Variable length arrows with custom styling

### ISSUE THREAD

**Issue by adigitoleo** [GenericMappingTools/GMT.jl#676]
Hi again,

This is really just a question, but it is specific to these bindings I think. Let me know if you'd prefer for this kind of thing to be posted to the forum instead.

I'm trying to plot variable length vectors on a map, to show plate motion. I can use `arrows` and put the length in the fourth column of the input data, however I also want to style the arrows (fill color, head size, ...). So, I thought about using the `arrow = (length = ...,)` argument, but I think it only accepts a single value (tried to give it an array anyway but got a core dump). Is there a way to do variable length arrows with custom vector styling using this API?

Cheers

**Comment by joa-quim** [comment]
Yes, for questions is preferable to use the forum. Please post a minimum example that shows the core dump. Crashes should never happen but when they do it becomes a pure GMT issue.

Regarding the solution, I'm afraid there is no ready to use GMT.jl recipe. What is needed is a multi-segment file with the arrow attributes in the multi-segment header (after the ``>`` sign) and the vector parameters in the data line. Note that this how it works with plain GMT. In GMT.jl we would need to come out with some function that fills the ``GMTdatasets`` headers (with data passed in from the higher level syntax), much at the image of what the ``make_zvals_vec()`` function does for choroplets maps.

**Comment by adigitoleo** [comment]
No problem, I'll check the GMT docs for how to set up the files. I'm not able to reproduce the crash so it was possibly unrelated. I'll keep learning GMT, hopefully I'll be able to offer more meaningful discussion at some point :)
